packages/openhim/src/Adaptor.js

```diff
@@ -28,38 +36,374 @@ export function execute(...operations) {
+ * @example <caption>Create a channel</caption>
+ * createChannel({
+ *   type: 'http',
+ *   authType: 'public',
+ *   status: 'enabled',
+ *   routes: [
+ *     {
+ *       name: 'FHIR Server Testing',
+ *       secured: false,
+ *       host: 'localhost',
+ *       port: '3447',
+ *       path: '',
+ *       pathTransform: '',
+ *       primary: true,
+ *       username: '',
+ *       password: '',
+ *       type: 'http',
+ *       status: 'enabled',
+ *       forwardAuthHeader: false,
+ *     },
+ *   ],
+ *   requestBody: true,
+ *   responseBody: true,
+ *   rewriteUrlsConfig: [],
+ *   rewriteUrls: false,
+ *   addAutoRewriteRules: true,
+ *   name: 'FHIR Server Testing',
+ *   urlPattern: '^/fhir/.*$',
+ *   methods: [
+ *     'GET',
+ *     'POST',
+ *     'DELETE',
+ *     'PUT',
+ *     'OPTIONS',
+ *     'HEAD',
+ *     'TRACE',
+ *     'CONNECT',
+ *     'PATCH',
+ *   ],
+ * });
+ * @function
+ * @public
+ * @param {object} body - The channel data to create.
+ * @returns {Operation}
+ * @state {HttpState}
+ */
+
+export function createChannel(body) {
+  return async state => {
+    const [resolvedBody] = expandReferences(state, body);
+    const response = await util.request(
+      state.configuration,
+      'POST',
+      '/channels',
+      {
+        body: resolvedBody,
+        parseAs: 'text',
+      }
+    );
+
+    return util.prepareNextState(state, response);
+  };
+}
+
+/**
+ * Get tasks query options
+ * @typedef {Object} OpenHIMGetTasksOptions
+ * @public
+ * @property {string} taskId - The ID of the task to retrieve. If provided, only this task will be returned.
+ * @property {number} filterLimit - The maximum number of tasks to return. Defaults to 100.
+ * @property {number} filterPage - The page to return (used in conjunction with filterLimit).
+ * @property {object} filters - Advanced filters to apply to the tasks. This is a JSON object that can include properties like `response.status` or `properties.prop`.
+ */
+
+/**
+ * Make a request to OpenHIM to get all tasks
+ * @example <caption>Get all tasks</caption>
+ * getTasks({
+ *   filterLimit: 10,
+ *   filterPage: 0,
+ *   filters: '{}',
+ * });
+ * @example <caption>Get a specific task</caption>
+ * getTasks({
+ *   taskId: '6870fbf470b851d7a22e9f05',
+ *   filterLimit: 10,
+ *   filterPage: 0,
+ *   filters: '{}',
+ * });
+ * @function
+ * @public
+ * @param {OpenHIMGetTasksOptions} options - Required request options for each request. See [OpenHIM Tasks docs](https://openhim.org/docs/api/tasks/read/#read-all-tasks)
+ * @returns {Operation}
+ * @state {HttpState}
+ */
+
+export function getTasks(options) {
+  return async state => {
+    const [resolvedoptions] = expandReferences(state, options);
+
+    const { taskId } = resolvedoptions;
+    delete resolvedoptions.taskId;
+
+    const response = await util.request(
+      state.configuration,
+      'GET',
+      taskId ? `/tasks/${taskId}` : '/tasks',
+      {
+        query: {
```

Comment:
A small thing but we don't actually need to spread this: just do `query: resolvedOptions`